src/App.js

```diff
@@ -14,11 +16,74 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    bool: false,
+    disabled: false,
+    options: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+  }
+
+  limit = (e) => {
+    const { target } = e;
+
+    this.setState(state => ({
+      goods: [...state.goods].filter(good => good.length >= target.value),
```

Comment:
```suggestion
      goods: state.goods.filter(good => good.length >= target.value),
```
filter и так делает копию